src/backend/index.js

```diff
@@ -126,6 +126,9 @@ function scan () {
         baseVue = baseVue.super
       }
       if (baseVue.config && baseVue.config.devtools) {
+        // give a unique id to root instance so we can
+        // 'namespace' its children
+        instance.__VUE_DEVTOOLS_ROOT_UID__ = rootInstances.length
```

Comment:
the `scan` function may get called multiple times (when the user force refreshes), in this case `rootInstances.length` may not be stable and existing root instances' ids may get mutated.

I think it's safer to have an incrementing value and only assign root UIDs when the root instance doesn't already have one:

``` js
// top level
let rootUID = 0

// ...
if (instance.__VUE_DEVTOOLS_ROOT_UID__ == null) {
  instance.__VUE_DEVTOOLS_ROOT_UID__ = rootUID++
}
```